packages/diagrams/frontend/sirius-components-diagrams/src/renderer/DiagramRenderer.tsx

```diff
@@ -154,6 +154,16 @@ export const DiagramRenderer = ({ diagramRefreshedEventPayload }: DiagramRendere
     }
   }, [diagramRefreshedEventPayload, diagramDescription]);
 
+  useEffect(() => {
+    setEdges((edges: Edge<EdgeData>[]) =>
+      edges.map((edge) => {
+        edge.updatable =
+          selection.entries.map((element) => element.id).includes(edge.data?.targetObjectId ?? '') && !readOnly;
+        return edge;
+      })
+    );
+  }, [selection]);
```

Comment:
You don't want an object in the dependency array here or it could lead to unnecessary rerenders when the value is the same. For example, clicking on the same item in the explorer will trigger a rerender, other scenario might trigger a rerender when we don't need one. Please note that it could also be avoided by memoising the selection in the `SelectionContext`.
Also note that we trigger a setEdges even if the selection change was not about edges, when selecting a lot of elements this might have some impacts. Maybe using `useOnSelectionChange` hook would allow for a more precise behavior.

Reply:
I changed this part to only active edge if it's selected and not if an element with same targeObjectId is selected